File: lab4/s2/server/src/main/java/ru/uvuv643/dto/human/ListHumanBeingDto.java
package ru.uvuv643.dto.human;

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;

import java.util.ArrayList;
import java.util.List;

@JacksonXmlRootElement(localName = "listHumanBeingDto") // Root XML tag
public class ListHumanBeingDto {

        @JacksonXmlElementWrapper(localName = "pairs") // Wrapper element for the list
        @JacksonXmlProperty(localName = "pairTeamHumanDto") // Inner tag for each object in the list
        private List<PairTeamHumanDto> humanBeingPair = new ArrayList<>();

        // Default Constructor
        public ListHumanBeingDto() {
        }

        // Constructor
        public ListHumanBeingDto(List<PairTeamHumanDto> humanBeingPair) {
                this.humanBeingPair = humanBeingPair;
        }

        // Getters and Setters
        public List<PairTeamHumanDto> getHumanBeingPair() {
                return humanBeingPair;
        }

        public void setHumanBeingPair(List<PairTeamHumanDto> humanBeingPair) {
                this.humanBeingPair = humanBeingPair;
        }
}

File: lab1/s2/src/main/kotlin/ru/uvuv643/soa/api/v1/dto/human/response/ListHumanBeingDto.kt
package ru.uvuv643.soa.api.v1.dto.human.response

import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlElementWrapper
import jakarta.xml.bind.annotation.XmlRootElement
import ru.uvuv643.soa.api.v1.dto.human.PairTeamHumanDto

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
data class ListHumanBeingDto(

        @XmlElementWrapper(name = "pairs")
        val humanBeingPair : List<PairTeamHumanDto>,

        ) {
        constructor() : this(listOf())
}